docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -52,52 +52,130 @@ const theme = createTheme({
 
 {{"demo": "GlobalThemeOverride.js"}}
 
-Each component is composed of several different parts.
-These parts correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
-You can use these classes inside the `styleOverrides` key to modify the corresponding parts of the component.
+Each component is composed of several different slots.
+These slots correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
+You can use these classes inside the `styleOverrides` key to modify the corresponding slots of the component.
+
+### Overrides based on props
```

Comment:
I don't have a great idea now, but I'm not sure "Overrides based on props" is the best title for this section, mostly because we're not just talking about _overrides_, we're also talking about _additions_ (adding a new prop to a component, which is different from changing something that already existed—an override).

Reply:
I'm changing to "Variants" based on @aarongarciah [comment](https://github.com/mui/material-ui/pull/42614#discussion_r1639729803).